### PR TITLE
Add remove method for LogStash::PLUGIN_REGISTRY #8316

### LINKED ISSUE #8316: LogStash::PLUGIN_REGISTRY should have a remove method
We can add a new plugin to the plugin registry, but there is no way to remove programmatically an existing plugin from the registry. This would be useful in testing or when a plugin takes over features of another plugin. This could be necessary for the transition to the universal plugin directory structure.

### PR DESCRIPTION
**Pull request by liketic**
Resolves #8316 . I'm new to `Ruby`, but happy to learn everything. Any comments are appreciated.  😄

### CODE REVIEW

**Comment by andrewvc** [comment]
jenkins, please test this

**Comment by jakelandis** [comment]
test failure un-related https://github.com/elastic/logstash/issues/8142

jenkins, please test this (again)

File: logstash-core/spec/logstash/plugins/registry_spec.rb [hunk 1]
```diff
@@ -61,6 +61,13 @@ def register; end
       expect(registry.lookup("filter", "simple_plugin")).to eq(simple_plugin)
     end
 
+    it "should plugin been removed" do
```
Comment (andrewvc) [inline logstash-core/spec/logstash/plugins/registry_spec.rb]: Can you change this to `it "should be possible to remove the plugin"`

Comment (liketic) [reply logstash-core/spec/logstash/plugins/registry_spec.rb]: Of course.

**Comment by andrewvc** [comment]
LGTM

**Comment by andrewvc** [comment]
Thanks @liketic !

**Pull request closed without merging**